### ISSUE TITLE
params 'missing'  due to rack bug

### ISSUE THREAD

**Issue by myitcv** [intridea/grape#559]
[This commit in `rack`](https://github.com/rack/rack/commit/b0593078ce792a380779528a6a135c066aa03515) fixes an issue whereby all requests which use a temp file (all requests larger 128 kb?) fail to be properly parsed.

The issue manifests itself in `grape` in the following way:

```ruby
params do
  optional :picture, type: String
end
patch '/' do
  picture_from_params = params[:picture] # nil - but should not be
  picture_from_env = env['rack.request.form_hash']['picture'] # not nil - as expected

  env["rack.request.form_input"].eql? env["rack.input"] # false - THIS IS THE BUG fixed by the commit in rack
  env["rack.request.form_input"].equal? env["rack.input"] # true
end
```

Indeed this could have been the root cause of a few issues raised in the Google Group. 

Not so much an issue from `grape`'s perspective apart from it being the place where this error manifests itself. Definitely something to be aware of when  running `Ruby 1.9+` (commit comment suggests this doesn't affect `Ruby 1.8.x`)

So more of an FYI in case anyone else gets bitten by this. 

The issue is further complicated by the fact that running `rack/test` doesn't cross this codepath. Presumably because temp files are not used in tests??

This issue is present in `rack == 1.5.2`, fixed in [`master`](https://github.com/rack/rack/tree/master) as of the time of this post, but not yet released (expected in `1.5.4` I believe)

**Comment by dblock** [comment]
Grape doesn't lock to a particular Rack version, so now that Rack 1.6.0 has shipped, this works. I committed the test with https://github.com/intridea/grape/commit/c72181dcb7d32e37dfd7a26084aeb6738cfa58a3.

**Comment by dblock** [comment]
Note that https://github.com/rack/rack/pull/756 back-ported the fix into Rack 1.5.3.

**Comment by TangMonk** [comment]
same problem
